pupil-spa/src/app/login/login.component.ts

```diff
@@ -81,14 +85,17 @@ export class LoginComponent implements OnInit, AfterViewInit {
         }
       },
       (err) => {
-        this.connectionFailed = err.status === 0;
-        this.loginSucceeded = false;
         this.submitted = false;
-        this.router.navigate(['sign-in']);
+        this.loginErrorService.changeMessage(err.message);
+        if (err.status === 401) {
+          this.loginSucceeded = false;
+          this.router.navigate(['sign-in']);
+        } else {
+          this.router.navigate(['sign-in-fail']);
```

Comment:
just to be clear - this is the page that surfaces the actual error detail?

Reply:
@GuyHarwood that is correct. Just to highlight the error message is being captured as an observable so there is no need to pass it down from the routing level.

Reply:
ah yes, i see that in the login failure component.  thanks for confirmation